src/types/type-declarations.ts

```diff
@@ -75,14 +79,20 @@ export interface GTag {
   (fn: 'config', opt: 'reset'): void;
 }
 
+export type DataLayer = Record<string, unknown>[];
+export type Maybe<T> = NonNullable<T> | undefined;
+
 export interface GoogleAnalyticsApi {
-  dataLayers: Record<string, Record<string, any>[]>;
-  gtag: GTag;
+  [key: string]: Maybe<DataLayer>;
```

Comment:
how to use it:

```
declare global {
    interface Window extends GoogleAnalyticsApi {}
}

const dataLayer = window["GADataLayer"];
dataLayer.push({"event": "login"})
```

Reply:
the above will successfully assume that dataLayer is of type DataLayer.

Reply:
WOuldn't this make any key possibly a DataLayer type ?

Reply:
right, any key could be a DataLayer type, since we don't know what names a user uses for one or multiple datalayers

Reply:
surely there's some ts magic we can do here using a generic, not sure how this works with the rest of the integration though so just use it for inspo 

```ts
export interface GoogleAnalyticsApi<T extends string = 'dataLayer'> {
  [key: T]: DataLayer;
}
```

Reply:
this seems quite dodgy with our existing implementation as it will augment the `window` types

surely there's some ts wizadry that can be done

Reply:
do you want to handle the type in nuxt/script, since removing the datalayer type would leave an empty type to export:

`export interface GoogleAnalyticsApi {}`

Reply:
We can handle it in nuxt scripts i guess. At least once https://github.com/nuxt/scripts/pull/176 is merged

Reply:
@huang-julien - this has been removed and I'm just exporting the DataLayer type now. 
The only outstanding question is, if you guys want the empty type exported, which I don't think makes really sense:
`export interface GoogleAnalyticsApi {}`

Reply:
we can remove it